### PR TITLE
fix bug: rserver cluster not found as spark cluster

### PR DESCRIPTION
**Pull request by voidfunction**

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java [hunk 1]
```diff
@@ -74,30 +74,29 @@ private ClusterManager() {
...
-                if (clusterDetail.getType().equals(type) && clusterDetail.getOSType().toLowerCase().equals(osType.toLowerCase())) {
-                    filterClusterDetailMap.put(clusterDetail.getName(), clusterDetail);
-                }
-            } else {
-                if (clusterDetail.getType().equals(type)) {
-                    filterClusterDetailMap.put(clusterDetail.getName(), clusterDetail);
-                }
+            ClusterType clusterType = clusterDetail.getType();
+            String myOsType =  clusterDetail.getOSType();
+            if (clusterType.equals(ClusterType.rserver) || clusterType.equals(ClusterType.spark)) {
```
Comment (ziyel) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java]: Do we need NPE check here?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java [hunk 2]
```diff
@@ -74,30 +74,29 @@ private ClusterManager() {
...
+
+                // remove Windows cluster
+               if (myOsType != null && osType != null && !myOsType.equalsIgnoreCase(osType)) {
+                   continue;
+               }
+                filterClusterDetailMap.put(clusterDetail.getName(), clusterDetail);
```
Comment (djlan) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java]: >  [](start = 15, length = 1)

NIT: remove extra space here

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java [hunk 3]
```diff
@@ -74,30 +74,29 @@ private ClusterManager() {
...
-            if (clusterDetail.getOSType() != null && osType != null) {
-                if (clusterDetail.getType().equals(type) && clusterDetail.getOSType().toLowerCase().equals(osType.toLowerCase())) {
-                    filterClusterDetailMap.put(clusterDetail.getName(), clusterDetail);
-                }
-            } else {
-                if (clusterDetail.getType().equals(type)) {
-                    filterClusterDetailMap.put(clusterDetail.getName(), clusterDetail);
-                }
+            ClusterType clusterType = clusterDetail.getType();
+            String myOsType =  clusterDetail.getOSType();
```
Comment (djlan) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java]: >  [](start = 30, length = 1)

NIT: remove extra space here

**Comment by djlan** [comment]
:shipit:

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java (hunk 1 above)
Comment (voidfunction) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java]: we have an `unknown` type for the mismatched type. so there's no NPE here.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java (hunk 3 above)
Comment (voidfunction) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java]: Fixed.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java (hunk 2 above)
Comment (voidfunction) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterManager.java]: Fixed.

**Pull request merged**